### DISCUSSION TITLE
页面标签中多页面地图显示不正常问题

### DISCUSSION THREAD

**Discussion by shihuixian** [BlazorComponent/MASA.Blazor#1509] [Q&A] [answered]
问题：在页面标签的不同页面中，使用leaflet初始化地图，只有一个地图能够显示正常，显示不正常的地图改变下浏览器窗口的大小就会显示正常。
使用的例子是官方提供的页面标签例子：
![image](https://github.com/masastack/MASA.Blazor/assets/40941172/72bfc3c6-c6f5-4edd-b94e-fc2729c67256)
重现步骤如下：
1.在每个页面中增加地图初始化代码，如下：
![image](https://github.com/masastack/MASA.Blazor/assets/40941172/6754cb5f-effa-48de-998c-62bd7a52d62e)
2._Host.html中增加如下内容：
`<link rel="stylesheet" href="https://unpkg.com/leaflet@1.9.4/dist/leaflet.css"
   integrity="sha256-p4NxAoJBhIIN+hmNHrzRCf9tD/miZyoHS5obTRR9BMY="
   crossorigin=""/>`
`<script src="https://unpkg.com/leaflet@1.9.4/dist/leaflet.js"
   integrity="sha256-20nQCchB9co0qIjJZRGuk2/Z9VM+kNiyxNV1lvTlZBo="
   crossorigin=""></script>`
`<script src="js/leafletfunction.js"></script>`
leafletfunction.js内容如下：
`function initMap(id) {
    var map = L.map(id).setView([51.505, -0.09], 13);
    L.tileLayer('https://tile.openstreetmap.org/{z}/{x}/{y}.png', {
    maxZoom: 19,
    attribution: '© OpenStreetMap'
}).addTo(map);
};`
显示不正常的地图，瓦片仅在左上角显示。

**Comment by capdiem** [comment]
@shihuixian 我尝试了一下，没有复现。

**Reply by shihuixian** [reply]
我这里是必现的，下面的代码是用模板生成的工程，我把上面的代码整理进来了，代码可通过下面链接下载：
https://pan.baidu.com/s/1S1SqsWD5mm5FaKwz84zXdg?pwd=ft7w
这种是显示不正常的：
![image](https://github.com/masastack/MASA.Blazor/assets/40941172/8218957c-9bed-4295-bed0-ebc6c0e751a7)
这种是正常的：
![image](https://github.com/masastack/MASA.Blazor/assets/40941172/45e6fefc-8924-4f19-83cf-45a71cb071b5)
OpenStreetMap要科学上网才能加载瓦片

**Reply by capdiem** [reply, marked as answer]
Masa.Blazor 1.0.2 是没问题的，1.0.1 可以复现。你升级到最新版吧。
